### ISSUE TITLE
`tf.keras.layers.experimental.preprocessing.Random*` fail in 0.0.1.dev220621

### ISSUE THREAD

**Issue by bobfromjapan** [microsoft/tensorflow-directml-plugin#217]
First of all, thanks for providing such a great plugin! I have been wanting to use DirectML acceleration with TensorFlow2 for some time now, and this plugin does it right.

I just noticed the existence of this plugin and immediately trained a simple image classification model with MobileNetV2 in the following environment.

- OS: Windows10 Home(21H2)
- CPU: Ryzen5 5600X(6C12T)
- MB: msi mpg B550 gaming plus
- RAM: 3200MHz 16GBx2
- GPU: RX6900XT
- Python: 3.9.7
- TensorFlow: 2.9.1

I installed the `tensorflow-directml-plugin` on a completely new Python virtual environment.

Then I got an error when running `tf.keras.layers.experimental.preprocessing.Random*`.

Maybe this issue is known to you, but I am reporting it. (I also understand that this plugin is still in preview)

Also, when I commented out the line containing `tf.keras.layers.experimental.preprocessing.Random*` and it ran with no problem. And, training become 10x faster than when trained on CPU only!

Full traceback is here: 

```
Traceback (most recent call last):
  File "C:\Users\Owner\dogandcat.py", line 85, in <module>
    history = model.fit(train_dataset,
  File "C:\Users\Owner\pyenvs\dml2\lib\site-packages\keras\utils\traceback_utils.py", line 67, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\Owner\pyenvs\dml2\lib\site-packages\tensorflow\python\eager\execute.py", line 54, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation model/sequential/random_rotation/stateful_uniform/RngReadAndSkip: Could not satisfy explicit device specification '' because the node {{colocation_node model/sequential/random_rotation/stateful_uniform/RngReadAndSkip}} was colocated with a group of nodes that required incompatible device '/job:localhost/replica:0/task:0/device:GPU:0'. All available devices [/job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:GPU:0].
Colocation Debug Info:
Colocation group had the following types and supported devices:
Root Member(assigned_device_name_index_=2 requested_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' assigned_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' resource_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' supported_device_types_=[CPU] possible_devices_=[]
_Arg: GPU CPU
RngReadAndSkip: CPU

Colocation members, user-requested devices, and framework assigned devices, if any:
  model_sequential_random_rotation_stateful_uniform_rngreadandskip_resource (_Arg)  framework assigned device=/job:localhost/replica:0/task:0/device:GPU:0
  model/sequential/random_rotation/stateful_uniform/RngReadAndSkip (RngReadAndSkip)

         [[{{node model/sequential/random_rotation/stateful_uniform/RngReadAndSkip}}]] [Op:__inference_train_function_13152]
```

<details>
<summary>FYI, the training code that caused the problem looks like this:</summary>

```
# Froked from https://github.com/tensorflow/docs-l10n/blob/master/site/ja/tutorials/images/transfer_learning.ipynb
import os
import tensorflow as tf

from tensorflow.keras.preprocessing import image_dataset_from_directory

_URL = 'https://storage.googleapis.com/mledu-datasets/cats_and_dogs_filtered.zip'
path_to_zip = tf.keras.utils.get_file('cats_and_dogs.zip', origin=_URL, extract=True)
PATH = os.path.join(os.path.dirname(path_to_zip), 'cats_and_dogs_filtered')

train_dir = os.path.join(PATH, 'train')
validation_dir = os.path.join(PATH, 'validation')

BATCH_SIZE = 32
IMG_SIZE = (160, 160)

DEVICE = '/GPU:0'

with tf.device(DEVICE):
  train_dataset = image_dataset_from_directory(train_dir,
                                              shuffle=True,
                                              batch_size=BATCH_SIZE,
                                              image_size=IMG_SIZE)

  validation_dataset = image_dataset_from_directory(validation_dir,
                                                    shuffle=True,
                                                    batch_size=BATCH_SIZE,
                                                    image_size=IMG_SIZE)

  val_batches = tf.data.experimental.cardinality(validation_dataset)
  test_dataset = validation_dataset.take(val_batches // 5)
  validation_dataset = validation_dataset.skip(val_batches // 5)

  AUTOTUNE = tf.data.AUTOTUNE

  train_dataset = train_dataset.prefetch(buffer_size=AUTOTUNE)
  validation_dataset = validation_dataset.prefetch(buffer_size=AUTOTUNE)
  test_dataset = test_dataset.prefetch(buffer_size=AUTOTUNE)

  data_augmentation = tf.keras.Sequential([
    tf.keras.layers.experimental.preprocessing.RandomFlip('horizontal'),
    tf.keras.layers.experimental.preprocessing.RandomRotation(0.2),
  ])

  preprocess_input = tf.keras.applications.mobilenet_v2.preprocess_input

  # Create the base model from the pre-trained model MobileNet V2
  IMG_SHAPE = IMG_SIZE + (3,)
  base_model = tf.keras.applications.MobileNetV2(input_shape=IMG_SHAPE,
                                                include_top=False,
                                                weights='imagenet')

  image_batch, label_batch = next(iter(train_dataset))
  feature_batch = base_model(image_batch)
  print(feature_batch.shape)

  base_model.trainable = True

  # Let's take a look at the base model architecture
  base_model.summary()

  global_average_layer = tf.keras.layers.GlobalAveragePooling2D()
  feature_batch_average = global_average_layer(feature_batch)
  print(feature_batch_average.shape)

  prediction_layer = tf.keras.layers.Dense(1)
  prediction_batch = prediction_layer(feature_batch_average)
  print(prediction_batch.shape)

  inputs = tf.keras.Input(shape=(160, 160, 3))
  x = data_augmentation(inputs)
  x = preprocess_input(x)
  x = base_model(x, training=False)
  x = global_average_layer(x)
  x = tf.keras.layers.Dropout(0.2)(x)
  outputs = prediction_layer(x)
  model = tf.keras.Model(inputs, outputs)

  base_learning_rate = 0.0001
  model.compile(optimizer=tf.keras.optimizers.Adam(lr=base_learning_rate),
                loss=tf.keras.losses.BinaryCrossentropy(from_logits=True),
                metrics=['accuracy'])

  model.summary()

  history = model.fit(train_dataset,
                      epochs=10,
                      validation_data=validation_dataset)
```
</details>

**Comment by PatriceVignola** [comment]
For the preview, we were focused on having roughly the same operator coverage as [tensorflow-directml](https://github.com/microsoft/tensorflow-directml), which is a fork of TensorFlow 1.15 and doesn't have support for `RngReadAndSkip`.

We'll definitely look into this operator and see if it makes sense to add support for it though. Issues like this help us prioritize which operators we want to add next, so thank you for bringing it to our attention!

**Comment by PatriceVignola** [comment]
Hi @bobfromjapan,

We just released [version 0.1.0.dev220928](https://pypi.org/project/tensorflow-directml-plugin/0.1.0.dev220928/) which should fix this problem.

**Comment by bobfromjapan** [comment]
I installed `tensorflow_directml_plugin-0.1.0.dev220928`, ran the same script, and confirmed that the problem does not occur.
Thanks for responding! I will close this issue.
